### PR TITLE
feat(init): #RDV-5 init repo with models, helpers, etc

### PR DESCRIPTION
**Pull request by niko-amosse**
## Describe your changes

- Initialized the repository with the core components for the project.
- Added the initial data models necessary for the early development phase.
- Included utility files (helpers) to simplify and standardize common operations.

## Issue ticket number and link

[RDV-5](https://jira.support-ent.fr/browse/RDV-5)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)
- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by Mattox0 — APPROVED** [review]

File: backend/src/main/java/fr/openent/appointments/model/IModel.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    /**
+     * @deprecated Should use IModelHelper instead
+     */
+    @Deprecated
+    default JsonArray toJsonArray(List<I> models) {
+        if (models == null) return null;
+        return new JsonArray(models.stream().map(IModel::toJson).collect(Collectors.toList()));
+    }
```
Comment (qperie) [inline backend/src/main/java/fr/openent/appointments/model/IModel.java:43]: A supprimer, c'est deprecated

File: backend/src/main/java/fr/openent/appointments/model/TransactionElement.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    // Functions
+
+    @Override
+    public JsonObject toJson() {
+        return new JsonObject()
+                .put("action", this.action.getValue())
+                .put("statement", this.query)
+                .put("values", this.params);
+    }
```
Comment (qperie) [inline backend/src/main/java/fr/openent/appointments/model/TransactionElement.java:81]: Magic string

File: backend/src/main/java/fr/openent/appointments/model/payload/GridPayload.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            .getJsonArray(Fields.CAMEL_TARGET_PUBLIC_LIST_ID, new JsonArray())
+            .stream()
+            .map(Object::toString)
+            .collect(Collectors.toList());
+
+        this.dailySlots = grid
+            .getJsonArray(Fields.CAMEL_DAILY_SLOTS, new JsonArray())
+            .stream()
+            .map(o -> new DailySlot((JsonObject) o))
+            .collect(Collectors.toList());
```
Comment (qperie) [inline backend/src/main/java/fr/openent/appointments/model/payload/GridPayload.java:55]: Tu peux utiliser IModelHelper

File: backend/src/main/java/fr/openent/appointments/core/constants/Fields.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
+package fr.openent.appointments.core.constants;
```
[the remark this answers is not in the document]
Comment (niko-amosse) [reply backend/src/main/java/fr/openent/appointments/core/constants/Fields.java:1]: fais pour tous les enums

File: backend/src/main/java/fr/openent/appointments/enums/Actions.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    this.value = value;
+  }
+
+  public String getValue() {
+    return this.value;
+  }
+
+  public static Actions getAction(String value) {
+     return Arrays.stream(Actions.values())
+             .filter(choiceTypes -> choiceTypes.getValue().equals(value))
```
[the remark this answers is not in the document]
Comment (niko-amosse) [reply backend/src/main/java/fr/openent/appointments/enums/Actions.java:25]: fais

File: backend/src/main/java/fr/openent/appointments/enums/Day.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    public static Day from(String value) {
+        for (Day type : values()) {
+            if (type.value.equals(value)) {
+                return type;
+            }
+        }
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (niko-amosse) [reply backend/src/main/java/fr/openent/appointments/enums/Day.java:29]: fais pour tous les enums

File: backend/src/main/java/fr/openent/appointments/model/payload/GridPayload.java (hunk 3 above)
Comment (niko-amosse) [reply backend/src/main/java/fr/openent/appointments/model/payload/GridPayload.java]: fait

**Review by qperie — APPROVED** [review]

**Pull request merged**